lib/bson/open_struct.rb

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2009-2014 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module BSON
+  # Injects behaviour for encoding OpenStruct objects using hashes
+  # to # raw bytes as specified by the BSON spec.
+  #
+  # @see http://bsonspec.org/#/specification
+  #
+  # @since 2.0.0
+  module OpenStruct
+
+    # Get the OpenStruct as encoded BSON.
+    #
+    # @example Get the hash as encoded BSON.
+    #   OpenStruct.new({ "field" => "value" }).to_bson
+    #
+    # @return [ String ] The encoded string.
+    #
+    # @see http://bsonspec.org/#/specification
+    #
+    # @since 2.0.0
+    def to_bson(buffer = ByteBuffer.new, validating_keys = Config.validating_keys?)
+      if RUBY_VERSION < "2.0.0"
```

Comment:
also we need to check for JRuby here. JRuby < 9.0 always returns nil from OpenStruct#to_h

```ruby
jruby-1.7.22 :003 > o = OpenStruct.new(:country => "Australia")
 => #<OpenStruct country="Australia"> 
jruby-1.7.22 :004 > o.to_h
 => nil 
```

JRuby 9.0:

```ruby
jruby-9.1.2.0 :003 > o = OpenStruct.new(:country => "Australia")
 => #<OpenStruct country="Australia"> 
jruby-9.1.2.0 :004 > o.to_h
 => {:country=>"Australia"} 
```